crates/extensions/c8y_mapper_ext/src/operations/handlers/custom_operation.rs

```diff
@@ -15,45 +20,110 @@ impl OperationContext {
             .context("Could not parse command as a custom operation command")
         {
             Ok(command) => command,
-            Err(_) => return Ok(OperationOutcome::Ignored),
+            Err(_) => return (Ok(OperationOutcome::Ignored), None),
         };
 
-        let sm_topic = &target.smartrest_publish_topic;
+        let mapper_id = self.command_id.prefix();
+        let operation = match self
+            .get_c8y_operation(mapper_id, &command.payload)
+            .context("Could not get CumulocitySupportedOperation from payload")
+        {
+            Ok(c8y_operation) => c8y_operation,
+            Err(_) => return (Ok(OperationOutcome::Ignored), None),
+        };
 
-        match command.get_command_status() {
+        let sm_topic = &target.smartrest_publish_topic;
+        let result = match command.get_command_status() {
             CommandStatus::Executing => Ok(OperationOutcome::Executing {
                 extra_messages: vec![],
             }),
             CommandStatus::Successful => {
-                let smartrest_set_operation = self.get_smartrest_successful_status_payload(
-                    CumulocitySupportedOperations::C8yCustom(operation_name.to_string()),
-                    cmd_id,
-                );
+                let smartrest_set_operation =
+                    self.convert_output(mapper_id, &command, operation.clone(), cmd_id);
                 let c8y_notification = MqttMessage::new(sm_topic, smartrest_set_operation);
 
                 Ok(OperationOutcome::Finished {
                     messages: vec![c8y_notification],
                 })
             }
             CommandStatus::Failed { reason } => {
-                let smartrest_set_operation = self.get_smartrest_failed_status_payload(
-                    CumulocitySupportedOperations::C8yCustom(operation_name.to_string()),
-                    &reason,
-                    cmd_id,
-                );
+                let smartrest_set_operation =
+                    self.get_smartrest_failed_status_payload(operation.clone(), &reason, cmd_id);
                 let c8y_notification = MqttMessage::new(sm_topic, smartrest_set_operation);
 
                 Ok(OperationOutcome::Finished {
                     messages: vec![c8y_notification],
                 })
             }
             _ => Ok(OperationOutcome::Ignored),
+        };
+
+        (result, Some(operation))
+    }
+
+    fn get_c8y_operation(
+        &self,
+        mapper_id: &str,
+        value: &serde_json::Value,
+    ) -> Option<CumulocitySupportedOperations> {
+        if let Some(value) = value.get(mapper_id) {
+            if let Some(c8y_operation) = value.get("on_fragment") {
+                if let Some(c8y_operation) = c8y_operation.as_str() {
+                    return Some(CumulocitySupportedOperations::C8yCustom(
+                        c8y_operation.to_string(),
+                    ));
+                }
+            }
+        }
+        None
```

Comment:
It's simpler to use [`Value::pointer()`](https://docs.rs/serde_json/latest/serde_json/value/enum.Value.html#method.pointer)

Reply:
Addressed in 43172ff512acc90477998570d800e38e4b060795